src/apps/tezos/sign_baker_op.py

```diff
@@ -0,0 +1,83 @@
+from trezor import wire, loop
+from trezor.crypto import hashlib
+from trezor.crypto.curve import ed25519
+from trezor.messages.TezosSignedBakerOp import TezosSignedBakerOp
+
+from apps.common import paths
+from apps.tezos import helpers, layout, writers
+from apps.tezos.writers import (
+    write_bool,
+    write_bytes,
+    write_uint8,
+    write_uint16,
+    write_uint32,
+    write_uint64,
+)
+
+
+async def sign_baker_op(ctx, msg, keychain):
+    paths.validate_path(ctx, helpers.validate_full_path, path=msg.address_n)
+    node = keychain.derive(msg.address_n, helpers.TEZOS_CURVE)
+
+    if helpers.check_baking_confirmed():
+        # Accept endorsements and block headers only, otherwise lock the device
+        if _check_operation_watermark(msg.magic_byte):
+            w = bytearray()
+            _get_operation_bytes(w, msg)
+            sig_prefixed = await _sign(ctx, bytes(w), node, msg)
+        else:
+            await helpers.prompt_pin()
```

Comment:
I'm not sure about this part. It will just throw you out of the staking mode when you send an "invalid" operation? Why not have the same behavior when you send a non-staking instruction?

My gut instinct would be to raise a wire error but stay in staking mode. But I'd like to talk a little more of the security implications.
ISTM that the only reason for sending something weird *in a baking operation* is a wire error, in which case the proper thing to do is retry.
OTOH someone might be trying to bust the one allowed operation?
But as long as they're busting, they could bust the other-operation-protection.
I dunno.

Reply:
> OTOH someone might be trying to bust the one allowed operation?
But as long as they're busting, they could bust the other-operation-protection.

After discussion, we concluded that this check is indeed redundant for the reason you mentioned above.